File: SOAP_DOTNET/CLI_WEB/src/main/java/ec/edu/espe/comercializadora/models/Factura.java
package ec.edu.espe.comercializadora.models;

public class Factura {
    private int id;
    private int clienteId;
    private String nombreCliente;
    private String cedulaCliente;
    private String fechaEmision;
    private double subtotal;
    private double iva;
    private double total;
    private String formaPago;
    private int cantidadProductos;

    public Factura() {}

    public Factura(int id, int clienteId, String nombreCliente, String cedulaCliente, String fechaEmision, 
                   double subtotal, double iva, double total, String formaPago, int cantidadProductos) {
        this.id = id;
        this.clienteId = clienteId;
        this.nombreCliente = nombreCliente;
        this.cedulaCliente = cedulaCliente;
        this.fechaEmision = fechaEmision;
        this.subtotal = subtotal;
        this.iva = iva;
        this.total = total;
        this.formaPago = formaPago;
        this.cantidadProductos = cantidadProductos;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getClienteId() {
        return clienteId;
    }

    public void setClienteId(int clienteId) {
        this.clienteId = clienteId;
    }

    public String getNombreCliente() {
        return nombreCliente;
    }

    public void setNombreCliente(String nombreCliente) {
        this.nombreCliente = nombreCliente;
    }

    public String getCedulaCliente() {
        return cedulaCliente;
    }

    public void setCedulaCliente(String cedulaCliente) {
        this.cedulaCliente = cedulaCliente;
    }

    public String getFechaEmision() {
        return fechaEmision;
    }

    public void setFechaEmision(String fechaEmision) {
        this.fechaEmision = fechaEmision;
    }

    public double getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(double subtotal) {
        this.subtotal = subtotal;
    }

    public double getIva() {
        return iva;
    }

    public void setIva(double iva) {
        this.iva = iva;
    }

    public double getTotal() {
        return total;
    }

    public void setTotal(double total) {
        this.total = total;
    }

    public String getFormaPago() {
        return formaPago;
    }

    public void setFormaPago(String formaPago) {
        this.formaPago = formaPago;
    }

    public int getCantidadProductos() {
        return cantidadProductos;
    }

    public void setCantidadProductos(int cantidadProductos) {
        this.cantidadProductos = cantidadProductos;
    }
}

File: SOAP_DOTNET/03 CLIMOV-comer/app/src/main/java/ec/edu/espe/comercializadora/models/Factura.kt
package ec.edu.espe.comercializadora.models

data class Factura(
    val id: Int = 0,
    val clienteId: Int = 0,
    val nombreCliente: String = "",
    val cedulaCliente: String = "",
    val fechaEmision: String = "",
    val subtotal: Double = 0.0,
    val iva: Double = 0.0,
    val total: Double = 0.0,
    val formaPago: String = "Efectivo", // "Efectivo" o "Crédito"
    val cantidadProductos: Int = 0
)

data class DetalleFactura(
    val id: Int = 0,
    val facturaId: Int = 0,
    val electrodomesticoId: Int = 0,
    val cantidad: Int = 1,
    val precioUnitario: Double = 0.0,
    val subtotal: Double = 0.0,
    val electrodomestico: Electrodomestico? = null
)

data class ElegibilidadCredito(
    val esElegible: Boolean = false,
    val mensaje: String = "",
    val montoMaximoCredito: Double = 0.0
)

data class FacturaRequest(
    val cedulaCliente: String = "",
    val detalles: List<DetalleFacturaRequest> = emptyList()
)

data class DetalleFacturaRequest(
    val electrodomesticoId: Int = 0,
    val cantidad: Int = 1
)

data class FacturaResponse(
    val exitoso: Boolean = false,
    val mensaje: String = "",
    val facturaId: Int? = null,
    val creditoId: Int? = null,
    val factura: Factura? = null
)
